src/features/defi/providers/cosmos/components/CosmosManager/Withdraw/CosmosWithdraw.tsx

```diff
@@ -0,0 +1,148 @@
+  const translate = useTranslate()
+  const [state, dispatch] = useReducer(reducer, initialState)
+  const { query, history, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId, contractAddress, assetReference } = query
+
+  const assetNamespace = 'slip44' // TODO: add to query, why do we hardcode this?
+  // Asset info
+  const assetId = toAssetId({
+    chainId,
+    assetNamespace,
+    assetReference, // TODO: handle multiple denoms
+  })
+  const underlyingAssetId = toAssetId({
+    // TODO: Underlying asset is the same as the staked asset for now, handle multiple denoms
+    chainId,
+    assetNamespace,
+    assetReference,
+  })
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const underlyingAsset = useAppSelector(state => selectAssetById(state, underlyingAssetId))
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
+  const feeAssetId = toAssetId({
+    chainId,
+    assetNamespace,
+    assetReference,
+  })
+  const feeMarketData = useAppSelector(state => selectMarketDataById(state, feeAssetId))
+
+  // user info
+  const chainAdapterManager = getChainAdapterManager()
+  const chainAdapter = chainAdapterManager.get(chainId)
+  const { state: walletState } = useWallet()
+
+  const opportunities = useCosmosSdkStakingBalances({ assetId })
+  const cosmosOpportunity = useMemo(
+    () =>
+      opportunities?.cosmosSdkStakingOpportunities?.find(
+        opportunity => opportunity.address === contractAddress,
+      ) ?? {},
+    [opportunities, contractAddress],
+  ) as unknown as MergedActiveStakingOpportunity // TODO: remove casting
+
+  useEffect(() => {
+    ;(async () => {
+      try {
+        if (!(walletState.wallet && contractAddress && chainAdapter)) return
+        const address = await chainAdapter.getAddress({ wallet: walletState.wallet })
+
+        dispatch({
+          type: CosmosWithdrawActionType.SET_USER_ADDRESS,
+          payload: address,
+        })
+        dispatch({
+          type: CosmosWithdrawActionType.SET_OPPORTUNITY,
+          payload: { ...cosmosOpportunity },
+        })
+      } catch (error) {
+        // TODO: handle client side errors
+        console.error('CosmosWithdraw error:', error)
+      }
+    })()
+  }, [cosmosOpportunity, chainAdapter, contractAddress, walletState.wallet])
+
+  const StepConfig: DefiStepProps = useMemo(() => {
+    return {
+      [DefiStep.Info]: {
+        label: translate('defi.steps.withdraw.info.title'),
+        description: translate('defi.steps.withdraw.info.yieldyDescription', {
+          asset: underlyingAsset?.symbol ?? '',
+        }),
+        component: Withdraw,
+      },
+      [DefiStep.Confirm]: {
+        label: translate('defi.steps.confirm.title'),
+        component: Confirm,
+      },
+      [DefiStep.Status]: {
+        label: 'Status',
+        component: Status,
+      },
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
add translate?

Reply:
2084b94812d2355172df46771e56ff3903ab300e

Reply:
react-hooks/exhaustive-deps removed in https://github.com/shapeshift/web/pull/2324